pkg/core/template/uri_loader.go

```diff
@@ -0,0 +1,65 @@
+package template
+
+import (
+	"net/url"
+
+	"github.com/skygeario/skygear-server/pkg/core/errors"
+)
+
+var ErrUnknownURIScheme = errors.New("unknown URI scheme")
+
+type URILoader struct {
+	EnableFileLoader bool
+	FileLoader       *FileLoader
+	EnableDataLoader bool
+	DataLoader       *DataLoader
+	AssetGearLoader  *AssetGearLoader
+}
+
+func NewURILoader(assetGearLoader *AssetGearLoader) *URILoader {
+	return &URILoader{
+		FileLoader:      &FileLoader{},
+		DataLoader:      &DataLoader{},
+		AssetGearLoader: assetGearLoader,
+	}
+}
+
+func (l *URILoader) Clone() *URILoader {
+	cloned := *l
+	return &cloned
+}
+
+func (l *URILoader) Load(uri string) (templateContent string, err error) {
+	u, err := url.Parse(uri)
+	if err != nil {
+		err = errors.HandledWithMessage(err, "failed to parse template URI")
+		return
+	}
+
+	switch u.Scheme {
```

Comment:
It seems that:
- `EnableFileLoader` is configurable through environment variable
- `EnableDataLoader` is always disabled currently
- `AssetGearLoader` is provided if appropriate environment variable is set (endpoint & master key)

Is that correct?

Reply:
Yes. DataLoader was introduced to implement test sender. But test sender was removed later. FileLoader is not intended for use on cloud deployment. It is helpful during testing. AssetGearLoader is for cloud deployment but is optional for local development.

Reply:
I suppose we'll have no API for sending test message, and if needed can use `DataLoader` to implement it in other project.

Reply:
Correct.